crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -178,6 +179,34 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter {
     type ExtraValueArgs = SuiExtraValueArgs;
     type Subcommand = SuiSubcommand<Self::ExtraValueArgs, Self::ExtraRunArgs>;
 
+    fn render_command_input(
+        &self,
+        task: &TaskInput<
+            TaskCommand<
+                Self::ExtraInitArgs,
+                Self::ExtraPublishArgs,
+                Self::ExtraValueArgs,
+                Self::ExtraRunArgs,
+                Self::Subcommand,
+            >,
+        >,
+    ) -> Option<String> {
+        match &task.command {
+            TaskCommand::Init(_, _)
+            | TaskCommand::PrintBytecode(_)
+            | TaskCommand::Publish(_, _)
+            | TaskCommand::Run(_, _) => None,
+            TaskCommand::Subcommand(SuiSubcommand::ProgrammableTransaction(..)) => {
+                let data_str = std::fs::read_to_string(task.data.as_ref()?)
+                    .ok()?
+                    .trim()
+                    .to_string();
+                Some(format!("{}\n{}", task.task_text, data_str))
+            }
+            TaskCommand::Subcommand(..) => None,
```

Comment:
nit: Re-order to avoid an extra case?

```suggestion
            TaskCommand::Subcommand(SuiSubcommand::ProgrammableTransaction(..)) => {
                let data_str = std::fs::read_to_string(task.data.as_ref()?)
                    .ok()?
                    .trim()
                    .to_string();
                Some(format!("{}\n{}", task.task_text, data_str))
            }
            TaskCommand::Init(_, _)
            | TaskCommand::PrintBytecode(_)
            | TaskCommand::Publish(_, _)
            | TaskCommand::Run(_, _)
            | TaskCommand::Subcommand(..) => None,
```